### PR TITLE
 Add str.replace (#133) and Knuth–Morris–Pratt version of str.contains.

### PR DESCRIPTION
**Pull request by radoslav11**
Also str.count(, regex=False) was implemented, however this doesn't follow the signature of the pandas version.

### CODE REVIEW

**Comment by xhochy** [comment]
Please install and use the pre-commit hooks to format the code,

**Comment by xhochy** [comment]
The string pattern generation is nice but that is something that could be done really well with [`hypothesis`](https://hypothesis.readthedocs.io/en/latest/) as done in some of the other tests. Can you have a look at it and try to add your code as `strategies` to it?

File: .gitignore [hunk 1]
```diff
@@ -109,3 +109,5 @@ ENV/
 
 # experiments
 private_*
+
+df.parquet
```
Comment (xhochy) [inline .gitignore:113]: Use the same as in #160 to avoid merge conflicts.
```suggestion
*.parquet
```

**Comment by xhochy** [comment]
Things to do before this can be merged:

* Add some benchmarks to `fletcher/benchmarks/benchmarks.py`
* Fix the failing CI builds

**Comment by Alaxe** [comment]
I have restricted some types of test cases in [tests/test_text.py:51][1] due to what appears to be a bug in pandas replace:
```
>>> pd.Series(['']).str.replace('', 'abc', n=1)
0
dtype: object
```
The correct result can be observed when running without n=1:
```
>>> pd.Series(['']).str.replace('', 'abc')
0    abc
dtype: object
```
[1]: https://github.com/xhochy/fletcher/blob/4c3eef3cb3a67ee811b88c20ec02cfeabd0de2d1/tests/test_text.py#L51

**Comment by Alaxe** [comment]
Added benchmarks.

There was a severe performance issue due to having a function call in the main loop of Knuth-Morris-Pratt for `append_character_to_kmp_matching`. Since using `@njit(inline='always')` didn't work reliably on my machine, I resolved this by manually inlining the function.

**Comment by xhochy** [comment]
@Alaxe Can you rebase again? Then this would be the next PR to go in 🥇

**Comment by xhochy** [comment]
Unit tests fail as you need to replace the usage of `fr_text` with `fr_strx`.

**Comment by Alaxe** [comment]
Fixed the unit tests.

There is, however, a slight decrease in code test coverage, should I look into fixing that?

File: tests/test_text.py [hunk 2]
```diff
@@ -53,18 +111,32 @@ def test_text_cat(data, fletcher_variant, fletcher_variant_2):
...
+def _check_str_to_t(t, func, data, fletcher_variant, test_offset=0, *args, **kwargs):
+    """Check a .str. function that returns a series with type t."""
+    tail_len = len(data) - test_offset
 
+    ser_pd = pd.Series(data, dtype=str).tail(tail_len)
     result_pd = getattr(ser_pd.str, func)(*args, **kwargs)
+
+    ser_fr = _fr_series_from_data(data, fletcher_variant).tail(tail_len)
+    print(type(data))
+    print(ser_fr)
```
Comment (xhochy) [inline tests/test_text.py:128]:
```suggestion
```

**Comment by xhochy** [comment]
The coverage decrease is so minimal, that's acceptable.

**Comment by xhochy** [comment]
Thanks!

**Pull request merged**